src/Microsoft.AspNetCore.Server.IIS/Core/IO/WebSocketsAsyncIOEngine.cs

```diff
@@ -100,7 +112,10 @@ private void ThrowIfNotInitialized()
 
         public void Dispose()
         {
-            NativeMethods.HttpTryCancelIO(_handler);
+            lock (_contextLock)
```

Comment:
Why do you need to lock here?

Reply:
Locking all IO

Reply:
We don't know that it's safe to call HttpTryCancelIO overlapped with other calls on the HttpContext